zebra-consensus/src/block/subsidy/founders_reward.rs

```diff
@@ -58,4 +158,90 @@ mod test {
+        // Todo: explain this better and make sure it is working propertly.
+        // after blossom there is still time left for the next change in
+        // founder address but the remaining is calculated with the new formula.
+        let mut addresses = FOUNDERS_REWARD_ADDRESSES_MAINNET;
+        let mut first_change_after_blossom = 656_866;
+
+        if network == Network::Testnet {
+            addresses = FOUNDERS_REWARD_ADDRESSES_TESTNET;
+            first_change_after_blossom = 584_794;
+        }
+
+        let blossom_height = Blossom.activation_height(network).unwrap();
+        let canopy_height = Canopy.activation_height(network).unwrap();
+
+        // the index in the founders reward address array that will be active at height
+        let mut index = 0;
+
+        // from genesis to blossom the founder reward address changes at FOUNDER_ADDRESS_CHANGE_INTERVAL
+        for n in (1..blossom_height.0).step_by(founders_address_change_interval().0 as usize) {
+            assert_eq!(
+                founders_reward_address(Height(n), network)?,
+                Address::from_str(addresses[index as usize]).expect("an address")
+            );
+            index += 1;
+        }
+
+        assert_eq!(
+            founders_reward_address(Height(first_change_after_blossom), network)?,
+            Address::from_str(addresses[index as usize]).expect("an address")
+        );
+
+        // after the first change after blossom the addresses changes at FOUNDER_ADDRESS_CHANGE_INTERVAL * 2
+        for n in (first_change_after_blossom..canopy_height.0)
+            .step_by((founders_address_change_interval().0 * 2) as usize)
+        {
+            assert_eq!(
+                founders_reward_address(Height(n), network)?,
+                Address::from_str(addresses[index as usize]).expect("an address")
+            );
+            index += 1;
+        }
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_founders_address_count() -> Result<(), Report> {
+        assert_eq!(
+            FOUNDERS_REWARD_ADDRESSES_MAINNET.len() as u32,
+            FOUNDERS_ADDRESS_COUNT
+        );
+        assert_eq!(
+            FOUNDERS_REWARD_ADDRESSES_TESTNET.len() as u32,
+            FOUNDERS_ADDRESS_COUNT
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_founders_address_ceiling() -> Result<(), Report> {
+        // Test proves why `div_ceil` needs to be used.
+        // `truncate(n/d) + 1` is 1 more than `ceiling(n/d)` when `n/d` is an integer.
+        // Suppose `SLOW_START_SHIFT.0 + PRE_BLOSSOM_HALVING_INTERVAL.0 = 480_000` instead
+        // of current protocol defined value of `850_000`.
+        let numerator = 480_000;
+
+        // `truncate(n) + 1` will output the wrong result
+        assert_eq!((numerator / FOUNDERS_ADDRESS_COUNT) + 1, 10001);
+
+        // `div_ceil` will output the right thing
+        assert_eq!(div_ceil(numerator, FOUNDERS_ADDRESS_COUNT), 10000);
+
+        Ok(())
+    }
```

Comment:
Sorry, I wasn't very clear about what I wanted in this test.

Write a test that calls `founders_address_for_network`, which gives different results with `div_ceil`.

Reply:
Actually, this test isn't that important, we should catch any actual errors using the canopy block test vectors.

Reply:
So please don't spent too much time on it.

Reply:
Should i just remove the test case ?

Reply:
Yes, I think we will cover this test case using property tests.